packages/ses/src/module-load.js

```diff
@@ -315,47 +400,78 @@ export const load = async (
     compartment,
   );
 
-  /** @type {Set<Promise<undefined>>} */
-  const pendingJobs = new Set();
   /** @type {Map<object, Map<string, Promise<Record<any, any>>>>} */
   const moduleLoads = new Map();
-  /** @type {Array<Error>} */
-  const errors = [];
 
-  const dependencyLoaded = memoizedLoadWithErrorAnnotation(
+  const { enqueueJob, drainQueue } = asyncOverseer();
+
+  enqueueJob(memoizedLoadWithErrorAnnotation, [
     compartmentPrivateFields,
     moduleAliases,
     compartment,
     moduleSpecifier,
-    pendingJobs,
+    enqueueJob,
+    preferAsync,
     moduleLoads,
+  ]);
+
+  // Drain pending jobs queue and throw an aggregate error
+  const errors = await drainQueue();
+
+  throwAggregateError({
     errors,
+    errorPrefix: `Failed to load module ${q(moduleSpecifier)} in package ${q(
```

Comment:
This error message should say `in compartment`  to be consistent with the other error message in the same file, but the inconsistency was introduced ~3 years ago and I'm not taking that on in this PR